napari_skimage_regionprops/_regionprops.py

```diff
@@ -184,3 +196,371 @@ def ellipsoid_axis_lengths(table):
+   
+    large_numbers = False
+    # If not provided, create default suffixes
+    if not suffixes:
+        suffixes = ['-ch' + str(i) for i in range(n_channels)]
+    else:
+        suffixes = ['-' + suffix for suffix in suffixes]
+
+    def highest_overlap(regionmask, intensity_image,
+                        overlap_threshold=intersection_area_over_object_area):
+        """
+        Gets the label number with highest overlap with label in another image.
+        
+        This function masks a labeled image called 'intensity_image' with 
+        'regionmask' and calculates the frequency of pixel values in that
+        region. Disconsidering zeros (background), it returns the most frequent
+        value if it overcomes the 'overlap_threshold'. Otherwise, it returns 0.
+        In case of draws, it returns the first occurence. This function follows
+        the standards of skimage.regionprops 'extra_properties'.
+         
+        Parameters
+        ----------
+        regionmask : (M, N[,P]) ndarray
+            Label image (probe channel). Labels to be used as a mask.
+        intensity_image : (M, N[,P]) ndarray
+            Label image (reference channel). Labels to be measured using probe
+            channel as a mask.
+        
+        Returns
+        -------
+        value : int
+            Most frequent label number under regionmask, except 0, that
+            overcomes threshold. Otherwise, it returns 0.
+        """
+        if overlap_threshold == 0:
+            return 0
+        values, counts = np.unique(np.sort(intensity_image[regionmask]),
+                                   return_counts=True)
+        # Probabilities of belonging to a certain label or bg
+        probs = counts/np.sum(counts)
+        # If there is underlying bg, take it out
+        if values[0] == 0:
+            values = np.delete(values, 0)
+            probs = np.delete(probs, 0)
+
+        # if any label overlap probability is bigger than overlap_threshold
+        if (probs >= overlap_threshold).any():
+            # find label with highest overlap
+            # if equal frequency, return first occurence
+            index_max_overlap_prob = np.argmax(probs)
+            value = values[index_max_overlap_prob]
+        else:  # otherwise, highest allowed overlap is considered to be
+            # with background, i.e., object does not "belong" to any
+            #  other label and gets 0
+            value = 0
+        return value
+
+    # Measure properties of reference channel
+    if intensity_image is not None:
+        # If single intensity image was given, then use it
+        if intensity_image.shape[0]==1:
+            image = intensity_image[0]
+        # Otherwise use the corresponding intensity image
+        else:
+            image = intensity_image[ref_channel]
+    else:
+        image = None
+    ref_channel_props = regionprops_table(image=image,
+                                          labels=labels_array[ref_channel],
```

Comment:
Would it make the code easier if you pass the `reference_label_image` as separate parameter instead of treating the first element of the array different than all the other elements?